### PR TITLE
[stable8.2] make swift url type configurable

### PR DESCRIPTION
**Pull request by butonic**
Allows configuring the Interface, typically 'public, 'internal' or 'admin'

cc @felixboehm

### CODE REVIEW

**Comment by butonic** [comment]
Needs port to master

File: lib/private/files/objectstore/swift.php [hunk 1]
```diff
@@ -77,7 +77,11 @@ protected function init() {
 			$serviceName = $this->params['serviceName'];
 		}
 
-		$this->objectStoreService = $this->client->objectStoreService($serviceName, $this->params['region']);
+		$urlType = null;
+		if ($this->params['urlType']) {
```
Comment (DeepDiver1975) [inline lib/private/files/objectstore/swift.php]: please use isset()

**Comment by MorrisJobke** [comment]
> Needs port to master

first of all we bring stuff to master and then backport stuff :see_no_evil: ;)

**Comment by DeepDiver1975** [comment]
> first of all we bring stuff to master and then backport stuff :see_no_evil: ;)

indeed

**Comment by MorrisJobke** [comment]
@karlitschek Please approve this backport

**Comment by karlitschek** [comment]
please backport

**Comment by felixboehm** [comment]
This works nice. 
Background here: Openstack provides 3 Swift endpoints and ownCloud used the service name "public" by default, which is ssl secured. We had some certificate issues then and want to use the "internal" service without ssl on an internal network.

**Comment by DeepDiver1975** [comment]
@butonic please submit PR against master first - I'm uncertain about the state of the CI system on stable8.2 regarding swift - failing test look ugly

**Comment by DeepDiver1975** [comment]
not 8.2.2 from my perspective because not yet merged to master and both (master and this) pr fail on unit testing :-1:

**Comment by butonic** [comment]
looking into master pr

**Comment by DeepDiver1975** [comment]
moving to 8.2.3 - we are closing 8.2.2 tomorrow

**Comment by MorrisJobke** [comment]
Ping - the master PR is merged.

**Comment by DeepDiver1975** [comment]
:+1:

**Comment by carlaschroder** [comment]
What are the exact options, 'public, 'internal' or 'admin'? Are there any others?

**Comment by carlaschroder** [comment]
And which oC versions does this apply to?

**Comment by MorrisJobke** [comment]
> And which oC versions does this apply to?

bildschirmfoto 2016-01-08 um 18 26 38


8.2.3 ;) (see on the right sidebar)

**Comment by carlaschroder** [comment]
@MorrisJobke, bitter experience has taught me to not rely on the milestone :)

**Comment by MorrisJobke** [comment]
> @MorrisJobke, bitter experience has taught me to not rely on the milestone :)

Okay :)

**Pull request merged**